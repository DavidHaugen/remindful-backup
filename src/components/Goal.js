import React, {Component } from 'react'

export default class Goal extends Component{
  style = this.props.goal.complete ? 'line-through' : 'none currentcolor solid'

  redirect(goalId){
    this.props.history.push(`/my-goals/${goalId}`)
  }

  render(){

    return (
      <div className='goal' onClick={() => this.redirect(this.props.goal.id)}>
          <li style={{textDecoration: this.style}} ><p className="overflowEl">{this.props.goal.name}</p></li>
      </div>
    )
  }
} 