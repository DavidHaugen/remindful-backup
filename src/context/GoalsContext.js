import React, { Component } from 'react'


const GoalsContext = React.createContext({
  goals: [],
  error: null,
  setError: () => {},
  clearError: () => {},
  setGoals: () => {}
})

export default GoalsContext

export class GoalsProvider extends Component {
  state = {
    goals: [],
    error: null,
  };

  setGoals = goals => {
    this.setState({ goals })
  }

  setError = error => {
    console.error(error)
    this.setState({ error })
  }

  clearError = () => {
    this.setState({ error: null })
  }

  render() {
    const value = {
      goals: this.state.goals,
      error: this.state.error,
      setError: this.setError,
      clearError: this.clearError,
      setGoals: this.setGoals
    }

    return (
      <GoalsContext.Provider value={value}>
        {this.props.children}
      </GoalsContext.Provider>
    )
  }
}