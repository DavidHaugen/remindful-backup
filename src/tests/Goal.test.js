import React from 'react';
import {MemoryRouter} from 'react-router-dom'
import Goal from '../components/Goal'
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer'
import GoalsContext from '../context/GoalsContext';

describe('<Goal />', () => {
  it('Renders without crashing', () => {
      shallow(<MemoryRouter><Goal/></MemoryRouter>);
  })
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<MemoryRouter><Goal goal={{complete: false, id: 1, name: 'test'}}/></MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});