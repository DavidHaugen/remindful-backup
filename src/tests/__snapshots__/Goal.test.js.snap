// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Goal /> renders the UI as expected 1`] = `
<div
  className="goal"
  onClick={[Function]}
>
  <li
    style={
      Object {
        "textDecoration": "none currentcolor solid",
      }
    }
  >
    test
  </li>
</div>
`;
